libgstc/libgstc_socket.c

```diff
@@ -85,16 +98,18 @@ gstc_socket_new (const char *address, const unsigned int port,
   if (connect (self->socket, (struct sockaddr *) &self->server,
           sizeof (self->server)) < 0) {
     ret = GSTC_UNREACHABLE;
-    goto close_socket;
+    close (self->socket);
+    goto free_self;
+  }
+
+  if (!self->keep_connection_open) {
+    close (self->socket);
```

Comment:
Smells like a refactor is needed, closing the socket in several error handlers. Refactor to an output label.

Reply:
The socket is now only closed here and only if the keep_connection_open flag is false. An output label isn't used since that's the only case where closing it is needed, other errors are triggered before the socket is opened or if it failed opening the socket.